bin/logstash.lib.sh

```diff
@@ -131,13 +131,19 @@ setup_ruby() {
   VENDORED_JRUBY=
 }
 
+logging_opts() {
+    printf "%s" " -J-DLog4jLogEventFactory=org.logstash.log.CustomLogEventFactory"
+    printf "%s" " -J-DLog4jContextSelector=org.logstash.log.StructuredMessageContextSelector"
```

Comment:
I recommend you to move these 2 properties to `src/main/resources/log4j2.component.properties` because :
- it will avoid you to copy this configuration in `setup.bat` (for windows startup)
- it will let you remove class LogEventFactoryRule where you define a third time these properties

So, doing that, you avoid to repeat yourself 3 times ! => DRY POWER ! :)